### ISSUE TITLE
Removing all cloudtrail event selectors results in an error

### ISSUE THREAD

**Issue by lorengordon** [terraform-providers/terraform-provider-aws#6628]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

* Terraform v0.11.10

### Affected Resource(s)

* aws_cloudtrail

### Terraform Configuration Files

```hcl
resource "aws_cloudtrail" "this" {
  name                       = "${var.cloudtrail_name}"
  s3_bucket_name             = "${var.cloudtrail_bucket}"
  enable_log_file_validation = true
  is_multi_region_trail      = true
  tags                       = "${var.tags}"
}
```

### Expected Behavior

Terraform should successfully remove any event selectors that happen to be associated to the CloudTrail.

### Actual Behavior

`terraform apply` resulted in an error:

```
aws_cloudtrail.this: Modifying... (ID: cloudtrail)
  event_selector.#:                          "1" => "0"
  event_selector.0.data_resource.#:          "1" => "0"
  event_selector.0.data_resource.0.type:     "AWS::S3::Object" => ""
  event_selector.0.data_resource.0.values.#: "1" => "0"
  event_selector.0.data_resource.0.values.0: "arn:aws:s3:::dummy-bucket-43123/" => ""

Error: Error applying plan:

1 error(s) occurred:

* aws_cloudtrail.this: 1 error(s) occurred:

* aws_cloudtrail.this: Error set event selector on CloudTrail (cloudtrail): InvalidEventSelectorsException: Specify a valid number of selectors (1 to 5) for your trail
        status code: 400, request id: 54b6a6d8-b99b-4959-a104-b5538014e7d6
```

### Steps to Reproduce

1. `terraform apply` to create the cloudtrail
2. Add an event selector via the console
3. `terraform apply` to try to remove the event selector

**Comment by b-dean** [comment]
If you didn't specify any `event_selector` then why should it remove existing ones? The other resources that have exhaustive lists don't behave this way. If you don't specify any of the nested things it doesn't go delete others that exist. Like with `aws_security_group` and its `ingress` or `egress` blocks.

**Comment by lorengordon** [comment]
That's a good point. I'd be fine in this case if it just didn't fail, also. :)

**Comment by lorengordon** [comment]
A notation that supported exclusivity would be handy, though.

**Comment by brandonstevens** [comment]
I just ran into this issue trying to revert from enabling data events. After reviewing the HTTP calls made by the Console, the call to remove all event selectors includes a single event selector with no data sources. All event selector management is performed through a single API Call, [PutEventSelectors](https://docs.aws.amazon.com/awscloudtrail/latest/APIReference/API_PutEventSelectors.html).

I was able to use the following to effectively remove all event selectors from a trail:

```
resource "aws_cloudtrail" "this" {
  name                       = "${var.cloudtrail_name}"
  s3_bucket_name             = "${var.cloudtrail_bucket}"
  enable_log_file_validation = true
  is_multi_region_trail      = true

  event_selector {
    read_write_type           = "All"
    include_management_events = true
  }

  tags = "${var.tags}"
}
```

While there may be an opportunity for the provider to handle this better (e.g. setting the above event selector configuration as default), I'm going to provide feedback to AWS to see if they can handle it through the API.
